shard/src/ext/purge_command.py

```diff
@@ -1,41 +1,192 @@
+import discord
 from discord.ext import commands
-from contextlib import suppress
 
+import re
+from datetime import datetime, timedelta
 
-@commands.command()
-async def purge(ctx, *args):
+TIME_REGEX = re.compile(r"(\d+)([m|s])")
+
+
+@commands.group()
+async def purge(ctx):
     '''
     Purge a channel of a user's messages
-    optionally include member id, channel, or message limit
+
+    Usage: {prefix}purge id {message id} [true]
+    or: {prefix}purge time {XXm|XXs}
+    The first will purge all messages sent after the message corresponding to the
+    message id of the user who sent that message. If true is passed, then all messages
+    from now until that message will be deleted.
+
+    Only searches the past 500 messages sent in a channel for which to delete.
+    If more than 500 messages have been sent between the given message's id and
+    when this command is run, then all of that user's messages may not get
+    deleted.
+
+    The second usage will delete all messages in the channel in the past number of
+    minutes or seconds.
+
+    NOTE: If the original message is deleted before the bot deletes it, all messages
+    from the target message's author in the channel will be deleted depending on the
+    timing of when the original message was deleted.
```

Comment:
why not just extract the time the message was sent and make that the limit?

Reply:
Can you still get a message after it has been deleted?

Reply:
no but you can get the time it was sent from its id

Reply:
https://discord.com/developers/docs/reference#snowflakes-snowflake-id-format-structure-left-to-right